karton_sqlmap/karton_sqlmap.py

```diff
@@ -102,37 +112,151 @@ def run(self, current_task: Task) -> None:
                     with open(target_path) as f:
                         # The format of the target is:
                         # url (METHOD)  # sqlmap_command
-                        # e.g. http://127.0.0.1:8000/vuln.php?id=4 (GET)  # sqlmap.py --technique B -v 4 -u http://127.0.0.1:8000/ --crawl=1
+                        # e.g. http://127.0.0.1:8000/vuln.php?id=4 (GET)  # sqlmap.py --technique B -v 4 -u http://127.0.0.1:8000/
                         target, _ = f.read().split("#", 1)
                         target = target.strip()
 
                     with open(log_path) as f:
                         log = f.read().strip()
 
-                    message = f"Found SQL Injection in {target}"
-                    result["target"] = target
-                    result["log"] = log
+                    found_sql_injection = FoundSQLInjection(
+                        message=f"Found SQL Injection in {target}", target=target, log=log
+                    )
 
                     version_query = "SELECT SUBSTR(VERSION(), 1, 15)"
                     for information_name, sqlmap_options, find_in_output in [
-                        ("version", ["--sql-query", version_query], version_query),
-                        ("user", ["--current-user"], "current user"),
+                        ("extracted_version", ["--sql-query", version_query], version_query),
+                        ("extracted_user", ["--current-user"], "current user"),
                     ]:
                         try:
-                            result[information_name] = self._call_sqlmap(
-                                url, sqlmap_options, find_in_output, timeout_seconds=SQLI_ADDITIONAL_DATA_TIMEOUT
-                            )  # type: ignore
+                            setattr(
+                                found_sql_injection,
+                                information_name,
+                                self._call_sqlmap(
+                                    url, sqlmap_options, find_in_output, timeout_seconds=SQLI_ADDITIONAL_DATA_TIMEOUT
+                                ),
+                            )
                         except Exception:  # Whatever happens, we prefer to report SQLi without additional data than no SQLi
                             self.log.exception(f"Unable to obtain {information_name} via blind SQL injection")
+                    return found_sql_injection
+        return None
+
+    @staticmethod
+    def _expand_query_parameters_for_scanning(url: str) -> List[str]:
```

Comment:
I would like to see example in some sort of comment, code is not very self-explanatory